### DISCUSSION TITLE
Should I be able to create a resource for an existing model?

### DISCUSSION THREAD

**Discussion by DLUK44** [filamentphp/filament#2001] [Help] [answered]
I have some existing models in my Laravel 9 project, Student being one of them. I have installed Filament and then run

```artisan make:filament-resource Student```

This seems to create the files referenced [here](https://filamentphp.com/docs/2.x/admin/resources#getting-started)

but when I browse to http://localhost/admin/, I do not see Students listed on the left hand side (it would be good if the documentation included that check, I wasn't sure if I should be seeing it or not at this stage). When I browse to http://localhost/admin/users/students I get a 404.

If I run ```artisan make:filament-resource Punk```, I see Punks (which I don't have an existing model for) listed on the left side of http://localhost/admin/ and I can click on it and the link goes to http://localhost/admin/punks

I am then told Class "App\Models\Punk" is not found which is probably to be expected.

If I then create a Punk model in Laravel, the link stays on the left of the Admin Panel and I can click on it and see 'No records found" (until I seed some data of course).

Why do none of my existing models appear in the admin panel? Has anyone else seen this issue before?

**Comment by danharrin** [comment, marked as answer]
Without seeing any of your code, I bet that you have a StudentPolicy that is not returning `true` for `viewAny()`. If you're not using it, delete the class. Then it will appear in the sidebar.

http://localhost/admin/users/students should be http://localhost/admin/students, that will be the correct URL when you click on the sidebar menu item.

You need an existing model to use a resource, the `Class not found` is because of that.

**Reply by DLUK44** [reply]
You are correct about the StudentPolicy not returning ```true``` for ```viewAny()``` and you are correct that getting rid of this or making it return true resolves the issue. This is the default in Laravel when creating models with the ````--policy```` or ````--all```` option though so I expect I won't be the only one.

Thank you.
